### PR TITLE
Recommendations (Jetpack Assistant): add unit tests to main React components

### PR DESCRIPTION
**Pull request by rcanepa**
Related to 1200070516266605-as-1200070517042906

#### Changes proposed in this Pull Request:

* Few small changes to the markup for better accessibility (read inline comments).
* Add unit tests to cover the behavior of the Recommendation module main components: `SiteType`, `FeaturePrompt`, and `Summary`.

#### Jetpack product discussion

This doesn't bring changes to the product, it only adds unit tests to the Recommendations module.

#### Does this pull request change what data or activity we track or use?

No. It doesn't change what data or activity we track or use.

#### Testing instructions:

**Code review**
* Make sure code changes make sense to you and that tests actually cover the main logic of each tested component.

**Running tests**
* Go to the `projects/plugins/jetpack` directory (`cd projects/plugins/jetpack`).
* Execute `yarn test-gui`.
* Make sure all tests pass.

**Recommendation flow**
* Give a quick test to the whole Recommendation flow to make sure that the small changes made to markup didn't affect the original behavior of it.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/checkbox-answer/index.jsx [hunk 1]
```diff
@@ -19,10 +19,16 @@ import { getDataByKey, updateRecommendationsData } from 'state/recommendations';
...
 const CheckboxAnswerComponent = ( { answerKey, checked, info, title, updateCheckboxAnswer } ) => {
-	const toggleCheckbox = useCallback( () => {
-		const newCheckedValue = ! checked;
-		updateCheckboxAnswer( { [ answerKey ]: newCheckedValue } );
-	}, [ answerKey, checked, updateCheckboxAnswer ] );
+	const toggleCheckbox = useCallback(
+		e => {
+			if ( e.target.type !== 'checkbox' ) {
+				return;
+			}
```
Comment (rcanepa) [inline projects/plugins/jetpack/_inc/client/recommendations/prompts/checkbox-answer/index.jsx:26]: Clicking on a `<label>` also triggers this event handler and we don't want it to run twice. We only want to run it when the target is of type `checkbox`.

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/checkbox-answer/index.jsx [hunk 2]
```diff
@@ -44,9 +50,11 @@ const CheckboxAnswerComponent = ( { answerKey, checked, info, title, updateCheck
 			tabIndex={ 0 }
 		>
 			<div className="jp-checkbox-answer__checkbox">
-				<input type="checkbox" checked={ checked } tabIndex={ -1 } />
+				<input id={ answerKey } type="checkbox" defaultChecked={ checked } tabIndex={ -1 } />
 			</div>
-			<div className="jp-checkbox-answer__title">{ title }</div>
+			<label htmlFor={ answerKey } className="jp-checkbox-answer__title">
+				{ title }
+			</label>
```
Comment (rcanepa) [inline projects/plugins/jetpack/_inc/client/recommendations/prompts/checkbox-answer/index.jsx:57]: Replaced `<div>` with `<label>` for better accessibility and testability.

File: projects/plugins/jetpack/_inc/client/recommendations/summary/index.jsx [hunk 3]
```diff
@@ -72,30 +72,34 @@ const SummaryComponent = props => {
...
+								<em>
+									{ __(
+										'You didn’t enable any recommended features. To get the most out of Jetpack, enable some recommendations or explore all Jetpack features.',
+										'jetpack'
+									) }
+								</em>
+							</p>
+						) }
+					</div>
+				</section>
```
Comment (rcanepa) [inline projects/plugins/jetpack/_inc/client/recommendations/summary/index.jsx:93]: Wrapping the enabled features in a section for better accessibility.

File: projects/plugins/jetpack/_inc/client/recommendations/summary/index.jsx [hunk 4]
```diff
@@ -72,30 +72,34 @@ const SummaryComponent = props => {
...
+					<section aria-labelledby="skipped-recommendations">
+						<h2 id="skipped-recommendations">{ __( 'Recommendations skipped', 'jetpack' ) }</h2>
 						<div>
 							{ summaryFeatureSlugs.skipped.map( slug => (
-								<FeatureSummary featureSlug={ slug } />
+								<FeatureSummary key={ slug } featureSlug={ slug } />
 							) ) }
 						</div>
-					</>
+					</section>
```
Comment (rcanepa) [inline projects/plugins/jetpack/_inc/client/recommendations/summary/index.jsx:102]: Wrapping the skipped features in a section for better accessibility.

File: projects/plugins/jetpack/_inc/client/test/test-utils.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		return <Provider store={ store }>{ children }</Provider>;
+	}
+	return rtlRender( ui, { wrapper: Wrapper, ...renderOptions } );
+}
+
+// re-export everything
+export * from '@testing-library/react';
+
+// override render method
+export { render };
```
Comment (rcanepa) [inline projects/plugins/jetpack/_inc/client/test/test-utils.js:29]: This is a pretty standard utils file that is used to create a render function that provides a Redux store to the component being tested. In short, it makes testing connected components much easier.

**Comment by rcanepa** [comment]
I don't know how to fix the `All commits were linted before commit.` warning. It seems that when I rebase the PR, all pre-commit checks are skipped.

**Comment by anomiex** [comment]
There isn't really a good way to do it. As long as the various "Linting" CI checks are passing it should be ok to ignore it; the main thing potentially missing is the "prettier" reformatting of JS and JSON files, which might result in a future PR editing those files having extraneous formatting changes.

The best way I've come up with to clear the warning is to do an interactive rebase (`git rebase -i origin/master`), change every commit from "pick" to "reword", and edit the commit messages to remove the `[not verified]`.

**Comment by rcanepa** [comment]
Rebased to solve merge conflicts.

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/test/component.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+			expect( activateFeatureStub.callCount ).to.be.equal( 1 );
+
+			// Restore stubs
+			recordEventStub.restore();
+			addSelectedRecommendationStub.restore();
+			mapDispatchToPropsStub.restore();
+		} );
+
+		it( 'calls the right actions when a user clicks on the skip feature button', () => {
+			const DUMMY_ACTION = { type: 'dummy' };
```
Comment (adimoldovan) [inline projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/test/component.js:133]: This is already declared on line 42

```suggestion

```

**Review by adimoldovan — DISMISSED** [review]
Looks good to me! I only left a minor suggestion.

File: projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/test/component.js (hunk 6 above)
Comment (rcanepa) [reply projects/plugins/jetpack/_inc/client/recommendations/prompts/feature-prompt/test/component.js]: Good catch! I'm definitely removing that unneccesary `const` definition.

**Comment by rcanepa** [comment]
Hey @adimoldovan, I removed the duplicated `const` definition (and rebased the PR) but Github dismissed your approval. Can I have another one after all checks are completed?

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**